src/objects/strategy-cards/strategy-card.js

```diff
@@ -128,6 +49,136 @@ function createStrategyCardUi(card, widgetFactory, height, color) {
     }
 }
 
+/**
+ * Builder class for registering a new strategy card object by adding event
+ * handlers as well as their removal in case of the destruction of the object.
+ */
+class RegisterStrategyCardUI {
+    /**
+     *
+     * @param {object} [data]
+     * @param {GameObject} [data.card] The object instance of the strategy card
+     * @param {function} [data.widgetFactory] A factory creating the UI for each player
+     * @param {Integer} [data.height] Height of the UI for each player
+     * @param {Color} [data.color] Color of the UI for each player
+     * @param {function} [data.onStrategyCardPlayed] Event handler of the "Play" button pressed.
+     * This function only is required if the Play button should do more than just creating a UI.
+     * The whole UI creation should belong into the widgetFactory to separate concerns.
+     * @param {function} [data.onStrategyCardSelectionDone] Event handler for the closing event.
+     */
+    constructor(data) {
+        this._card = data?.card;
```

Comment:
Wait, `?.` is legal?  I thought we needed transpiling to use that syntax?

Reply:
nah only in the test environment... now I see some nasty errors in the game console. changed the logic